<a href="https://colab.research.google.com/github/Trisinus123/PembelajaranMesin_Sem3/blob/main/TaskPractice1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import pandas as pd

df5 = pd.read_csv('/content/wbc.csv')
df5.head()


id diagnosis  radius_mean  texture_mean  perimeter_mean  area_mean  \
0    842302         M        17.99         10.38          122.80     1001.0   
1    842517         M        20.57         17.77          132.90     1326.0   
2  84300903         M        19.69         21.25          130.00     1203.0   
3  84348301         M        11.42         20.38           77.58      386.1   
4  84358402         M        20.29         14.34          135.10     1297.0   

   smoothness_mean  compactness_mean  concavity_mean  concave points_mean  \
0          0.11840           0.27760          0.3001              0.14710   
1          0.08474           0.07864          0.0869              0.07017   
2          0.10960           0.15990          0.1974              0.12790   
3          0.14250           0.28390          0.2414              0.10520   
4          0.10030           0.13280          0.1980              0.10430   

   ...  texture_worst  perimeter_worst  area_worst  smoothness_worst  \
0  ...          17.33           184.60      2019.0            0.1622   
1  ...          23.41           158.80      1956.0            0.1238   
2  ...          25.53           152.50      1709.0            0.1444   
3  ...          26.50            98.87       567.7            0.2098   
4  ...          16.67           152.20      1575.0            0.1374   

   compactness_worst  concavity_worst  concave points_worst  symmetry_worst  \
0             0.6656           0.7119                0.2654          0.4601   
1             0.1866           0.2416                0.1860          0.2750   
2             0.4245           0.4504                0.2430          0.3613   
3             0.8663           0.6869                0.2575          0.6638   
4             0.2050           0.4000                0.1625          0.2364   

   fractal_dimension_worst  Unnamed: 32  
0                  0.11890          NaN  
1                  0.08902          NaN  
2                  0.08758          NaN  
3                  0.17300          NaN  
4                  0.07678          NaN  

[5 rows x 33 columns]

In [3]:
df5.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 569 entries, 0 to 568
Data columns (total 33 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   id                       569 non-null    int64  
 1   diagnosis                569 non-null    object 
 2   radius_mean              569 non-null    float64
 3   texture_mean             569 non-null    float64
 4   perimeter_mean           569 non-null    float64
 5   area_mean                569 non-null    float64
 6   smoothness_mean          569 non-null    float64
 7   compactness_mean         569 non-null    float64
 8   concavity_mean           569 non-null    float64
 9   concave points_mean      569 non-null    float64
 10  symmetry_mean            569 non-null    float64
 11  fractal_dimension_mean   569 non-null    float64
 12  radius_se                569 non-null    float64
 13  texture_se               569 non-null    float64
 14  perimeter_se             5

# Deskripsi Tugas

Pada tugas pratikum ini Anda akan menggunakan data "Wisconsin Breast Cancer". Data tersebut terdiri dari 569 data yang digunakan untuk mendiagnonis jenis kanker Malignant (M) dan Benign (B). Tugas Anda adalah,
Pisahkan antara variabel yang dapat digunakan dan variabel yang tidak dapat digunakan.

1. Pisahkan antara variabel yang dapat digunakan dan variabel yang tidak dapat digunakan.

2. Lakukan proses encoding pada kolom "diagnosis".

3. Lakukan proses standarisasi pada semua kolom yang memiliki nilai numerik.

4. Lakukan proses stratified split data untuk membuat data latih dan data uji dengan rasio 80:20.

In [4]:
# Jawaban 1
df5 = df5[['id', 'diagnosis', 'radius_mean', 'radius_se', 'radius_worst']]
df5.head()

id diagnosis  radius_mean  radius_se  radius_worst
0    842302         M        17.99     1.0950         25.38
1    842517         M        20.57     0.5435         24.99
2  84300903         M        19.69     0.7456         23.57
3  84348301         M        11.42     0.4956         14.91
4  84358402         M        20.29     0.7572         22.54

In [6]:
# Jawaban 2
from sklearn.preprocessing import LabelEncoder

le = LabelEncoder() # membuat objek dari LabelEncoder
df5['diagnosis'] = le.fit_transform(df5['diagnosis']) # proses encoding

df5.head()

id  diagnosis  radius_mean  radius_se  radius_worst
0    842302          1        17.99     1.0950         25.38
1    842517          1        20.57     0.5435         24.99
2  84300903          1        19.69     0.7456         23.57
3  84348301          1        11.42     0.4956         14.91
4  84358402          1        20.29     0.7572         22.54

In [11]:
# Jawaban 3
from sklearn.preprocessing import StandardScaler

std = StandardScaler()

# Pisahkan kolom numerik yang akan distandardisasi
kolom_numerik = df5.select_dtypes(include=['float64', 'int64'])


# Lakukan standarisasi pada kolom numerik
df5[kolom_numerik.columns] = std.fit_transform(kolom_numerik)

# Tampilkan DataFrame yang sudah distandardisasi
df5.head()


id  diagnosis  radius_mean  radius_se  radius_worst
0 -0.236405   1.297676     1.097064   2.489734      1.886690
1 -0.236403   1.297676     1.829821   0.499255      1.805927
2  0.431741   1.297676     1.579888   1.228676      1.511870
3  0.432121   1.297676    -0.768909   0.326373     -0.281464
4  0.432201   1.297676     1.750297   1.270543      1.298575

In [10]:
# Jawaban 4
from sklearn.model_selection import train_test_split

# Pisahkan kolom target (label) dari fitur-fitur
X = df5.drop(columns=['diagnosis'])
y = df5['diagnosis']

# Melakukan stratified split data dengan rasio 80:20
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, stratify=y, random_state=42)

# Menampilkan ukuran data latih dan data uji
print("Jumlah data latih:", len(X_train))
print("Jumlah data uji:", len(X_test))

Jumlah data latih: 455
Jumlah data uji: 114
